core/src/data/string_ext.rs

```diff
@@ -7,28 +7,68 @@ pub enum StringExtError {
 }
 
 pub trait StringExt {
-    fn like(&self, pattern: &str, case_sensitive: bool) -> Result<bool>;
+    fn like(&self, pattern: &str, case_sensitive: bool, escape_char: char) -> Result<bool>;
 }
 
 impl StringExt for str {
-    fn like(&self, pattern: &str, case_sensitive: bool) -> Result<bool> {
+    fn like(&self, pattern: &str, case_sensitive: bool, escape_char: char) -> Result<bool> {
         let (match_string, match_pattern) = match case_sensitive {
             true => (self.to_owned(), pattern.to_owned()),
             false => {
                 let lowercase_string = self.to_lowercase();
                 let lowercase_pattern = pattern.to_lowercase();
-
                 (lowercase_string, lowercase_pattern)
             }
         };
 
-        Ok(Regex::new(&format!(
-            "^{}$",
-            regex::escape(match_pattern.as_str())
-                .replace('%', ".*")
-                .replace('_', ".")
-        ))
-        .map_err(|_| StringExtError::UnreachablePatternParsing)?
-        .is_match(match_string.as_str()))
+        let mut regex_pattern = String::new();
+        let mut chars = match_pattern.chars().peekable();
+        let ignore_case = !case_sensitive;
+        let escaped_escape_char = regex::escape(&escape_char.to_string());
+
+        while let Some(current_char) = chars.next() {
+            if current_char == escape_char {
+                if let Some(next_char) = chars.peek() {
+                    if *next_char == escape_char
+                        || *next_char == '_'
+                        || *next_char == '%'
+                        || *next_char == '\''
+                    {
+                        regex_pattern.push(*next_char);
+                        chars.next();
+                    } else {
+                        regex_pattern.push_str(&escaped_escape_char);
+                    }
+                } else {
+                    regex_pattern.push_str(&escaped_escape_char);
+                }
+            } else {
+                match current_char {
+                    '_' => regex_pattern.push('.'),
+                    '%' => regex_pattern.push_str(".*"),
+                    '\'' => regex_pattern.push_str("\\'"),
+                    _ => regex_pattern.push(current_char),
+                }
+            }
+        }
+
+        let regex_pattern = if ignore_case {
+            format!("(?i){}", regex_pattern)
+        } else {
+            regex_pattern
+        };
+
+        let regex = match case_sensitive {
+            true => Regex::new(&format!("^{}$", regex_pattern)),
+            false => Regex::new(&format!("(?i)^{}$", regex_pattern)),
+        };
+
+        match regex {
+            Ok(regex) => {
+                let escaped_match_string = regex::escape(match_string.as_str());
+                Ok(regex.is_match(escaped_match_string.as_str()))
+            }
+            Err(_) => Err(StringExtError::UnreachablePatternParsing.into()),
+        }
     }
 }
```

Comment:
We are trying to avoid using `mut` keyword in GlueSQL engine codes.
This is my implementation.
```suggestion
    fn like(&self, pattern: &str, case_sensitive: bool, escape_char: char) -> Result<bool> {
        let pattern = pattern
            .chars()
            .scan(false, |escaped, char| {
                if !*escaped && char == escape_char {
                    *escaped = true;

                    return Some("".to_owned());
                }

                let regex = match (&escaped, &char) {
                    (false, '_') => ".".to_owned(),
                    (false, '%') => ".*".to_owned(),
                    (true, '\\') => "\\\\".to_owned(),
                    _ => char.to_string(),
                };

                *escaped = false;

                Some(regex)
            })
            .collect::<String>();

        let matched = RegexBuilder::new(&format!("^{pattern}$"))
            .case_insensitive(!case_sensitive)
            .build()
            .map_err(|_| StringExtError::UnreachablePatternParsing)?
            .is_match(self);

        Ok(matched)
    }
```